src/attention/utils/helpers.ts

```diff
@@ -1,98 +1,243 @@
+import {
+  computePosition,
+  flip,
+  offset,
+  shift,
+  arrow,
+  autoUpdate,
+  ReferenceElement,
+} from '@floating-ui/dom'
 
-const TOP = "top";
-const BOTTOM = "bottom";
-const LEFT = "left";
-const RIGHT = "right";
-type Directions = "top" | "right" | "bottom" | "left";
+const TOPSTART = 'top-start'
+const TOP = 'top'
+const TOPEND = 'top-end'
+const RIGHTSTART = 'right-start'
+const RIGHT = 'right'
+const RIGHTEND = 'right-end'
+const BOTTOMSTART = 'bottom-start'
+const BOTTOM = 'bottom'
+const BOTTOMEND = 'bottom-end'
+const LEFTSTART = 'left-start'
+const LEFT = 'left'
+const LEFTEND = 'left-end'
+
+type Directions =
+  | 'top'
+  | 'top-start'
+  | 'top-end'
+  | 'right'
+  | 'right-start'
+  | 'right-end'
+  | 'bottom'
+  | 'bottom-start'
+  | 'bottom-end'
+  | 'left'
+  | 'left-start'
+  | 'left-end'
 
 export const opposites = {
+  [TOPSTART]: BOTTOMEND,
   [TOP]: BOTTOM,
+  [TOPEND]: BOTTOMSTART,
+  [BOTTOMSTART]: TOPEND,
   [BOTTOM]: TOP,
+  [BOTTOMEND]: TOPSTART,
+  [LEFTSTART]: RIGHTEND,
   [LEFT]: RIGHT,
+  [LEFTEND]: RIGHTSTART,
+  [RIGHTSTART]: LEFTEND,
   [RIGHT]: LEFT,
-};
+  [RIGHTEND]: LEFTSTART,
+}
+
 export const arrowLabels = {
-  [TOP]: "↑",
-  [BOTTOM]: "↓",
-  [LEFT]: "←",
-  [RIGHT]: "→",
-};
-export const directions = [TOP, BOTTOM, LEFT, RIGHT];
-export const rotation = {
+  [TOPSTART]: '↑',
+  [TOP]: '↑',
+  [TOPEND]: '↑',
+  [BOTTOMSTART]: '↓',
+  [BOTTOM]: '↓',
+  [BOTTOMEND]: '↓',
+  [LEFTSTART]: '←',
+  [LEFT]: '←',
+  [LEFTEND]: '←',
+  [RIGHTSTART]: '→',
+  [RIGHT]: '→',
+  [RIGHTEND]: '→',
+}
+export const directions = [
+  TOPSTART,
+  TOP,
+  TOPEND,
+  BOTTOMSTART,
+  BOTTOM,
+  BOTTOMEND,
+  LEFTSTART,
+  LEFT,
+  LEFTEND,
+  RIGHTSTART,
+  RIGHT,
+  RIGHTEND,
+]
+export const rotation: any = {
+  [LEFTSTART]: -45,
   [LEFT]: -45,
+  [LEFTEND]: -45,
+  [TOPSTART]: 45,
   [TOP]: 45,
+  [TOPEND]: 45,
+  [RIGHTSTART]: 135,
   [RIGHT]: 135,
+  [RIGHTEND]: 135,
+  [BOTTOMSTART]: -135,
   [BOTTOM]: -135,
-};
+  [BOTTOMEND]: -135,
+}
 
 export type AttentionState = {
-  isShowing?: boolean;
-  isCallout?: boolean;
-  actualDirection?: Directions;
-  directionName: Directions;
-  arrowEl?: HTMLElement | null;
-  attentionEl?: HTMLElement | null;
-  targetEl?: unknown;
-  top?: Boolean;
-  right?: Boolean;
-  bottom?: Boolean;
-  left?: Boolean;
-  tooltip?: Boolean;
-  popover?: Boolean;
-  callout?: Boolean;
-  noArrow?: Boolean;
-  waitForDOM?: () => void;
-};
-
-const middlePosition = "calc(50% - 7px)";
-const isDirectionVertical = (name: string) => [TOP, BOTTOM].includes(name);
+  isShowing?: boolean
+  isCallout?: boolean
+  actualDirection?: Directions
+  directionName?: Directions
+  arrowEl?: HTMLElement | null
+  attentionEl?: HTMLElement | null
+  flip?: Boolean
+  fallbackPlacements?: Directions[]
+  targetEl?: unknown
+  tooltip?: Boolean
+  popover?: Boolean
+  callout?: Boolean
+  highlight?: Boolean
```

Comment:
It doesn't seem like these need to be defined in `AttentionState` type. At least I couldn't see them being used in any of the utils.

Reply:
Updated in commit: [6c52c9d](https://github.com/warp-ds/core/pull/15/commits/6c52c9dd87f76c84b60c6c03389dbcd4302f36c2)